jenkins-infra/jenkins-infra
# Automatic Update of Docker Engine using updatecli

Hello :wave: 

https://issues.jenkins.io/browse/INFRA-3074

Please find a first answer to this issue in that Pull Request allowing to:

- Check for the latest version of docker-ce with apt,
- Set that version in the Puppet's provisioning file.

Please note that for now, this PR requires updatecli to be executed on:

- A Debian or Ubuntu environment,
- With apt,
- With docker-ce GPG keys and repositories configured,

As per now, there are no particular verification done on the Puppet's provisionning file.

Here are some future improvements that could be planned around this PR:

- Relying on a Docker image for resolving the latest docker-ce version (allowing to run the script on any environment basically),
- Checking for the Puppet's provisionning file to exist (I believe it's not yet possible through conditions, but could be a thing)
- Check for the existence of docker::version key in the Puppet's provisionning file (https://github.com/updatecli/updatecli/issues/289 needed)

Please don't hesitate to let me know what you think :)
